src/transformers/generation_utils.py

```diff
@@ -891,55 +966,52 @@ def generate(
             return_dict_in_generate if return_dict_in_generate is not None else self.config.return_dict_in_generate
         )
 
+        if pad_token_id is None and eos_token_id is not None:
+            # special case if pad_token_id is not defined
+            logger.warning(f"Setting `pad_token_id` to `eos_token_id`:{eos_token_id} for open-end generation.")
+            pad_token_id = eos_token_id
+
+        # 2. Define model inputs
+        # inputs_tensor has to be defined
+        # model_input_name is defined if model-specific keyword input is passed
+        # otherwise model_input_name is None
+        # all model-specific keyword inputs are removed from `model_kwargs`
+        inputs_tensor, model_input_name = self._prepare_model_inputs(inputs, bos_token_id, model_kwargs)
```

Comment:
This is much nicer IMO than it was before because now we can be sure that `inputs_tensor` **cannot** be `None` anymore. We can also nicely wrap `inputs_embeds` into other possible encoder inputs here